lib/sequin/databases_runtime/slot_message_store.ex

```diff
@@ -260,109 +293,177 @@ defmodule Sequin.ConsumersRuntime.ConsumerMessageStore do
 
     schedule_flush(state)
 
-    case Consumers.get_consumer(state.consumer_id) do
-      {:ok, consumer} ->
-        messages = load_messages(consumer)
-        {:noreply, %{state | messages: messages, consumer: consumer}}
-
-      {:error, %Error.NotFoundError{entity: :consumer}} ->
-        Logger.error("[ConsumerMessageStore] Consumer #{state.consumer_id} not found")
-        {:stop, :normal, state}
+    case backfill(state) do
+      {:ok, state} -> {:noreply, state}
+      {:error, _} -> {:stop, :normal, state}
     end
   end
 
   @impl GenServer
   def handle_call({:put_messages, messages}, _from, %State{} = state) do
-    messages =
-      messages
-      |> Stream.map(fn msg -> %{msg | ack_id: UUID.uuid4(), dirty: true} end)
-      |> Map.new(&{&1.record_pks, &1})
+    {time, state} =
+      :timer.tc(fn ->
+        messages =
+          messages
+          # This could be dangerous in prod but is useful in test 🤔
+          |> Stream.map(fn
+            %{ack_id: ack_id} = msg when is_binary(ack_id) -> %{msg | dirty: true}
+            msg -> %{msg | ack_id: UUID.uuid4(), dirty: true}
+          end)
+          |> Map.new(&{&1.ack_id, &1})
+
+        _state = State.put_messages(state, messages)
+      end)
 
-    state = State.put_messages(state, messages)
+    Logger.info(
+      "[SlotMessageStore] Put #{length(messages)} messages in #{div(time, 1000)}ms (message_count=#{map_size(state.messages)})"
+    )
```

Comment:
Again this might become debug or cloud-only

Reply:
What do you think about adopting this pattern, now that we're always displaying metadata in the logs?

```suggestion
    Logger.info(
      "[SlotMessageStore] Put messages", count: map_size(state.messages), time_ms: div(time, 1000)"
    )
```

Should be friendly in dev and easy for machines to parse

Only downside I can think of is that metadata is not displayed in single-line in Datadog by default. But there's probably a way to navigate this...